packages/entities/entities-routes/README.md

```diff
@@ -15,6 +15,17 @@ Route entity components.
 - `@kong/kongponents` must be added as a dependency in the host application, globally available via the Vue Plugin installation, and the package's style imports must be added in the app entry file. [See here for instructions on installing Kongponents](https://kongponents.konghq.com/#globally-install-all-kongponents).
 - `@kong-ui-public/i18n` must be available as a `dependency` in the host application.
 - `axios` must be installed as a dependency in the host application
+- `@kong-ui-public/expressions` and `monaco-editor` must be installed as **`dependencies`** to support [Kong's Expressions language](https://docs.konghq.com/gateway/latest/reference/expressions-language/)-related features. If you don't need the Expressions support, install them as **`devDependencies`**, and add the following configuration to your Vite config to avoid bundling them with your build result:
+
+  ```ts
+  defineConfig({
+    build: {
+      rollupOptions: {
+        external: ['@kong-ui-public/expressions', 'monaco-editor'],
+      },
+    },
+  })
+  ```
```

Comment:
Why would a host app need to add this to the config? By adding to externals here in this repo, the packages will not be bundled, meaning they don't exist in the host app regardless. 

If there are imports from these packages here, they would need to be dependencies in the host app; I'm not sure it makes sense to "skip" these imports by telling the host app not to resolve them?

Reply:
As `entities-routes` contains components that conditionally import from `@kong-ui-public/expressions` (Expressions features). For example:

https://github.com/Kong/public-ui-components/blob/c7993636c84a3efebdbc3a38ba9e246b8bd37432/packages/entities/entities-routes/src/components/RouteFormExpressionsEditorLoader.vue#L62-L70

… which is used here:

https://github.com/Kong/public-ui-components/blob/c7993636c84a3efebdbc3a38ba9e246b8bd37432/packages/entities/entities-routes/src/components/RouteForm.vue#L338-L362

(Note: the Expressions features are not enabled by default because we want to keep the default behavior unchanged as much as possible)

However, if we import `RouteForm` in the host app but don't want to use the Expressions feature (setting `routeFlavors` without `expressions`) provided by `RouteForm`, `@kong-ui-public/expressions` and `monaco-editor` will still be bundled. For instance, in Konnect UI's `overview` app, we import `RouteForm`, but if we don't pass `routeFlavors` with `expressions` to it, it should not enable/use Expressions features. Specifying `@kong-ui-public/expressions` and `monaco-editor` in the `devDependencies` of the `overview` app still makes them be bundled, and will increase the bundle size by ~4MB:

<img width="1362" alt="Screenshot 2024-04-22 at 20 43 32" src="https://github.com/Kong/public-ui-components/assets/5277268/3eaeb8f0-d5b8-466a-bfbc-a60a80d09144">

Therefore, I would suggest the host apps using components (we only have `RouteForm` for now) that may import `@kong-ui-public/expressions` to mark them as `external` in their Vite config.

I hope this explanation helps you understand better.

Reply:
Good idea. I've moved them to `optionalDependencies` and updated the docs for `entities-routes`. Would it look better this time?